src/lib/core/base/base2.ts

```diff
@@ -15,9 +15,14 @@ import {buildLayoutCSS} from '../../utils/layout-validator';
 
 export abstract class BaseDirective2 implements OnChanges, OnDestroy {
 
+  private destroySubject: Subject<void> = new Subject();
```

Comment:
We discussed that we wouldn't do this because it adds complexity with very little benefit